include/circt/Dialect/Handshake/HandshakeOps.td

```diff
@@ -814,6 +814,32 @@ def JoinOp : Handshake_Op<"join", [
   let hasCustomAssemblyFormat = 1;
 }
 
+def SyncOp : Handshake_Op<"sync", [
+  HasClock, DeclareOpInterfaceMethods<ExecutableOpInterface>,
+  DeclareOpInterfaceMethods<GeneralOpInterface>,
+  AllTypesMatch<["operands", "results"]>
+]> {
+  let summary = "sync operation";
+  let description = [{
+    Produces valid outputs with the same data once all inputs become available.
```

Comment:
I think this is a bit vague.
Suggestion:

Synchronizes an arbitrary set of inputs. Synchronization implies applying join semantics in between all in- and output ports.